Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -412,6 +412,64 @@ and/or costly.
     of the SoC watchdog toward multiple operating systems, however it
     could go beyond that, and reach higher levels of complexity.
 
+  * **System Decomposition**
+ 
+    The previous points can be leveraged and extended to cases where the
+    safety requirements rely on multiple subsystems, possibly across different
+    software layers.
+    
+    As usual, it is extremely important to assess upfront the requirements
+    for availability.
+
+    A requirement for simply detecting the potential interference can be easier
+    to deal with, than a requirement to ensure continued safety operations,
+    in the presence of interference.
+
+    Leaving aside, for now, considerations about the latter scenario, the former
+    one still carries a need for the detection to happen within a maximum interval,
+    measured from the beginning of the interference.
+
+    With these two requirements in mind:
+    * detecting interference
+    * detecting the interference within a maximum allowed time
+      
+    One can attempt to isolate those subsystems directly involved, and investigate
+    the possibility of periodically validating its continued integrity.
+    
+    Here, the basic concept is that one could be able to provide - periodically -
+    a preset stimuli, and compare the live output of the system with the
+    pre-calculated expected one, that the system should produce, when free from
+    interference.
+    
+    Taking as example an image recognition pipeline, tasked with receiving an input
+    from a sensor and recognising items from a given set (e.g. a speed limit sign),
+    this could be implemented in the following way:
+    * periodically inject a known image, of a specific object(s)
+    * expect that within a given time the system will correctly identify the object(s)
+    
+    Such an approach has the advantage of not requiring any special partitioning of
+    the architecture (e.g. which parts must be implemented as user processes and
+    which parts must be kernel components).
+    
+    However, to work, it must also satisfy several conditions:
+    * the monitoring must happen form a context that is at least of equal ASIL level
+    * the monitoring context must not be affected by the interference it is supposed to detect
+    * the testing must have sufficient coverage for detecting all the relevant types of inteference
+    * the testing must account for any recently collected hystorical data
+      that the system might rely on.
+      (e.g. lighting statistics from past frames might get corrupted and affect
+      the following evaluations, but if they are not part of the periodic self test,
+      this corruption wil lgo unnoticed, because the system tested is different from
+      the system in real use)
+
+    The requirements of the Fault Deteting Time Interval (FTDI) and the time required
+    for performing a self-test will dictate how often this self test must be performed.
```

Comment:
I would not use the word "test" here. In general the last part of an ASIL Decomposition flow is a "comparator" comparing the outputs from the monitor and the monitored system respectively.
Also you should not confuse the FDTI with the comparator sampling interval. You may have also an analog continuous comparator that trigger an exception as soon as the delta reaches a certain threashold. In this regard the FDTI is the max time to handle the exception...

Reply:
I meant to indicate an upper bound on the priodicity. I will reword the text to make it more explicit. Then the analog comparator will be automatically included (although even that is not really istantaneous, it will have a specific delay of propagation - nothing is instantaneous).

Reply:
done